common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -683,6 +683,16 @@ public UTF8String trimRight() {
     return copyUTF8String(0, e);
   }
 
+  /**
+   * Return the length of string stripped at most `numOfSpaces` tailing spaces
+   */
+  public int effectiveLength(int numOfSpaces) {
```

Comment:
how about `public UTF8String trimTrailingSpaces(int numSpaces)`? I don't think we need to save the string creating for the error case, as perf doesn't matter much for that case.

Reply:
Considering that we need to calculate the new string's numChars again for error raising.

Reply:
ah good point!